conda/models/match_spec.py

```diff
@@ -183,7 +184,10 @@ def from_dist_str(cls, dist_str):
         if '::' in dist_str:
             channel_subdir_str, dist_str = dist_str.split("::", 1)
             if '/' in channel_subdir_str:
-                channel_str, subdir = channel_subdir_str.split('/', 2)
+                channel_str, subdir = channel_subdir_str.rsplit('/', 1)
+                if subdir not in context.known_subdirs:
+                    channel_str = channel_subdir_str
+                    subdir = None
                 parts.update({
                     'channel': channel_str,
                     'subdir': subdir,
```

Comment:
I'm a little nervous here about adding `parts["subdir"] = None`, and I don't think we need to. Instead of 

```python
parts.update({
    'channel': channel_str,
    'subdir': subdir,
})
```

let's just do

```python
parts["channel"] = channel_str
if subdir:
    parts["subdir"] = subdir
```

Reply:
Sounds good, I made that change